### ISSUE TITLE
Teams-for-linux segfault when trying to share screen on Wayland

### ISSUE THREAD

**Issue by dklementowski** [NixOS/nixpkgs#213058]
### Describe the bug
The application source is here: https://github.com/IsmaelMartinez/teams-for-linux
I don't think the problem is in upstream as the Flatpak version works fine and that's what I use for now. 
The application works, I can join a call, but it crashes when trying to share a screen. Maybe there's something missing, like access to xdg-desktop-portal? 

In the Terminal it throws:

```
[nix-shell:~]$ teams-for-linux 
[9054:0128/025903.862990:ERROR:gpu_init.cc(523)] Passthrough is not supported, GL is egl, ANGLE is 
Segmentation fault (zrzut pamięci)
```

### Steps To Reproduce
1. Run the application on Wayland
2. Try to share a screen
3. Expect crash

### Expected behavior
It should allow to share screen with PipeWire through xdg-desktop-portal.

### Additional context
I use Plasma on Wayland on AMD graphics. I don't experience issues with screen sharing in other apps like Slack or Firefox (installed from nixpkgs).
Let me know if I can help somehow with testing and what not.

### Notify maintainers
@muscaln 

### Metadata

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.8, NixOS, 23.05 (Stoat), 23.05.20230127.14b6cf7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.2`
 - channels(root): `"nixos"`
 - channels(dominik): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by lilyinstarlight** [comment]
I use xdg-desktop-portal-wlr and am able screen share to Teams for Linux just fine. I'll see if I can replicate this for Plasma in a VM or something in the next week or so

In the mean time, can you post the output of `strace -ff -e trace=openat 2>&1 | grep -F .so` so I can see if it's failing to load some required library?

**Comment by dklementowski** [comment]
> I use xdg-desktop-portal-wlr and am able screen share to Teams for Linux just fine. I'll see if I can replicate this for Plasma in a VM or something in the next week or so
> 
> In the mean time, can you post the output of `strace -ff -e trace=openat teams-for-linux 2>&1 | grep -F .so` so I can see if it's failing to load some required library?

Sure https://gist.github.com/dklementowski/d7cc6083b647bc2e4fdbdeabcc977273

**Comment by lilyinstarlight** [comment]
It looks like it couldn't find libpulse.so even though #205790 was merged. Are you sure you're using a version built after that PR was merged?

Also I will be updating the package to 1.0.49 which may help as well, but it'll have to wait on #214062 because of fun Yarn stuff

**Comment by dklementowski** [comment]
> It looks like it couldn't find libpulse.so even though #205790 was merged. Are you sure you're using a version built after that PR was merged?

I guess... I'm on nixpkgs-unstable and I update pretty frequently. I can check that if Ill find a moment.

> Also I will be updating the package to 1.0.49 which may help as well, but it'll have to wait on #214062 because of fun Yarn stuff

No rush for me as the Flatpak version does the thing. Maybe it'll be fixed with a new build

**Comment by lilyinstarlight** [comment]
> I guess... I'm on nixpkgs-unstable and I update pretty frequently. I can check that if Ill find a moment.

It's also possible it just seg faulted before it found it (I did simple parsing to extract all .so names it looked for and all names it found and did a diff to see what was never found, rather than doing something more context-aware)

I'll look some more sometime in the next week and see if I can replicate the issue in a VM (and if not maybe it has something to do with AMD hardware acceleration)

**Comment by dklementowski** [comment]
Ok, I can now confirm that it was my fault and the problem was fixed in the meantime.
I tried to run the app using `nix-shell -p teams-for-linux` and I didn't have channels configured properly to get fresh packages. In fact it was always getting the older version from my local cache.

I close the issue, the app now works perfectly! Thanks for your assistance and sorry for wasting your time, I'm still new to Nix.
